modules/account/src/main/kotlin/edu/stanford/spezi/module/account/register/RegisterScreen.kt

```diff
@@ -105,50 +110,84 @@ fun RegisterScreen(
             style = titleSmall,
         )
         VerticalSpacer(height = Spacings.large)
-        Text("CREDENTIALS", style = labelLarge)
-        ValidatedOutlinedTextField(
-            modifier = Modifier.fillMaxWidth(),
-            value = uiState.email.value,
-            onValueChange = {
-                onAction(Action.TextFieldUpdate(it, TextFieldType.EMAIL))
-            },
-            labelText = "E-Mail Address",
-            errorText = uiState.email.error,
-        )
+        IconLeadingContent(
+            icon = Icons.Outlined.Email,
+            content = {
+                ValidatedOutlinedTextField(
+                    modifier = Modifier.fillMaxWidth(),
+                    value = uiState.email.value,
+                    onValueChange = {
+                        onAction(Action.TextFieldUpdate(it, TextFieldType.EMAIL))
+                    },
+                    labelText = "E-Mail Address",
+                    errorText = uiState.email.error,
+                )
+            })
         VerticalSpacer(height = Spacings.small)
         if (!uiState.isGoogleSignUp) {
-            ValidatedOutlinedTextField(
-                modifier = Modifier.fillMaxWidth(),
-                value = uiState.password.value,
-                onValueChange = {
-                    onAction(Action.TextFieldUpdate(it, TextFieldType.PASSWORD))
-                },
-                labelText = "Password",
-                errorText = uiState.password.error,
-                visualTransformation = PasswordVisualTransformation(),
-            )
+            IconLeadingContent(
+                icon = Icons.Outlined.Lock,
+                content = {
+                    ValidatedOutlinedTextField(
+                        modifier = Modifier.fillMaxWidth(),
+                        value = uiState.password.value,
+                        onValueChange = {
+                            onAction(Action.TextFieldUpdate(it, TextFieldType.PASSWORD))
+                        },
+                        labelText = "Password",
+                        errorText = uiState.password.error,
+                        trailingIcon = {
+                            IconButton(onClick = { onAction(Action.TogglePasswordVisibility) }) {
+                                val icon: Painter = if (uiState.isPasswordVisible) {
```

Comment:
Minor: I would initialize `val iconId: Int` here and then invoke `painter = painterResource(id = iconId)` below